Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -10879,6 +10882,31 @@ def xsiam_api_call_with_retries(
     return response
 
 
+def split_data_to_chunks(data, chunk_size):
+    """
+    Splits a string of data into chunks of a specified size.
+
+    :type data: ``str``
+    :param data: The string of data to split.
+    :type chunk_size: ``int``
+    :param chunk_size: The maximum size of each chunk.
+
+    :return: A list of chunks containing the data.
+    :rtype: ``list``
+    """
+    chunk = []
+    curr_size = 0
+    for data_part in data.split('\n'):
+        chunk.append(data_part)
+        curr_size += sys.getsizeof(data_part)
+        if curr_size > chunk_size:
```

Comment:
If you check the size after the addition, the size will always be greater than the `target_chunk_size` (except for the last iteration).
Let's switch the order a bit, and instead check the size before addition - if it's greater, then `yield` the data, if not then add it and continue.